pkg/operator/staticpod/controller/staticpodstate/staticpodstate_controller.go

```diff
@@ -127,10 +132,8 @@ func (c *StaticPodStateController) sync(ctx context.Context, syncCtx factory.Syn
 
 	switch {
 	case len(images) == 0:
-		// if NodeStatuses is still empty, we are most probably in bootstrapping phase and this controller races with the
-		// installer controller. Hence, ignore that case. It will settle.
-
-		if len(originalOperatorStatus.NodeStatuses) > 0 {
+		// only generate an event if we found pods, but none has an image.
```

Comment:
wondering what the case "we found pods, some have an image, some don't" means. A pod without image is strange.

Reply:
yeah
```
cat emptypod.yaml 
apiVersion: v1
kind: Pod
metadata:
  name: emptypod
  namespace: default
$ kubectl apply -f emptypod.yaml -v2
The Pod "emptypod" is invalid: spec.containers: Required value
```

Reply:
and
```
apiVersion: v1
kind: Pod
metadata:
  name: testemt
  namespace: default
spec:
  containers:
  - image: ""
    imagePullPolicy: Always
    name: test2
kubectl apply -f emptypod.yaml -v2
The Pod "testemt" is invalid: spec.containers[0].image: Required value
```